Blog post for 2020-02-14 (#18)
puppetlabs/iac: _posts/2020-02-14-status-update.md

---
layout: post
title: "14th Feb 2020: IAC Team Status Update"
author: criskoduck
categories:
  - team
  - status
tags:
  - module updates
  - tooling
  - personal
  - honeycomb
  - metrics
  - litmus
  - windows
  - powershell
---

> The IAC team have been working on a lot of interesting things these past two weeks. Here's a brief overview of what we've been up to:

Reviewer: ```suggestion
The IAC team has been working on a lot of interesting things over the past week. Here's a brief overview of what we've been up to:
```

Author: Copy / pasta - d'oh!

> ## Updates to Puppet Modules

Reviewer 2: Technically we released two! pwshlib went out last night with a bugfix for puppet server 5.x

Author: Cool - I'll add it. Kerminator didn't list it, unfortunately. 11 hours ago - must be below the cut off?

We've released an update to one module since last week's update: 
- [puppetlabs/scheduled_task](https://forge.puppet.com/puppetlabs/scheduled_task) (v2.0.1)

This update to [puppetlabs/scheduled_task](https://forge.puppet.com/puppetlabs/scheduled_task) resolves an issue reported in [MODULES-10101](https://tickets.puppetlabs.com/browse/MODULES-10101).
We'll have more module releases next week, but a few interesting updates to come:

> - **[puppetlabs-java](https://forge.puppet.com/puppetlabs/java):** Thanks to [timdeluxe](https://github.com/timdeluxe) for [the PR](https://github.com/puppetlabs/puppetlabs-java/pull/370) to add support for [AdoptOpen JDK](https://adoptopenjdk.net/) to the module. [fraenki](https://github.com/fraenki) was also a great help in getting this PR over the line. We'll releasing an update to the module soon!

Reviewer 2: Suggest a new line before a new list.

- **[puppetlabs-mysql](https://forge.puppet.com/puppetlabs/mysql):** [This is a nice PR](https://github.com/puppetlabs/puppetlabs-mysql/pull/1192) from [mauricemeyer](https://github.com/mauricemeyer) that now allows a user to override the default configuration for mysql, if you wish. Thanks to [bastelfreak](https://github.com/bastelfreak) for the help during the review stages.

> - **[puppetlabs-apt](https://forge.puppet.com/puppetlabs/apt):** [anarcat](https://github.com/anarcat) submitted [this fix](https://github.com/puppetlabs/puppetlabs-apt/pull/904) which also highlightd an issue with our tests too. Two for the price of one! Thanks to [bastelfreak](https://github.com/bastelfreak) and [baldurmen](https://github.com/baldurmen) for the reviews, too.

Reviewer 2: ```suggestion
- **[puppetlabs-apt](https://forge.puppet.com/puppetlabs/apt):** [anarcat](https://github.com/anarcat) submitted [this fix](https://github.com/puppetlabs/puppetlabs-apt/pull/904) which also highlighetd an issue with our tests. Two for the price of one! Thanks to [bastelfreak](https://github.com/bastelfreak) and [baldurmen](https://github.com/baldurmen) for the reviews, too.
```

> ## We're still buzzing about Honeycomb

Reviewer 2: Clever! 💜

Author: These awful bee puns are beginning to sting. 😆 🐝

![Module RSpec trace](/iac/assets/2020-02-14-status-update/honeycomb_rspec_formatter.png)

We're continuing to see the benefits of [Honeycomb](https://www.honeycomb.io/). To make life easier, we now have the [rspec_honeycomb_formatter](https://github.com/puppetlabs/rspec_honeycomb_formatter).

> When you add this gem to a project's Gemfile, it will create a [Custom formatter in RSpec](https://relishapp.com/rspec/rspec-core/docs/formatters/custom-formatters) that will capture the metrics we're interested in and then sends it to the configured Honeycomb endpoint for processing!

Reviewer 2: ```suggestion
When you add this gem to a project's Gemfile, it will create a [Custom formatter in RSpec](https://relishapp.com/rspec/rspec-core/docs/formatters/custom-formatters) that will capture the metrics we're interested in and then send it to the configured Honeycomb endpoint for processing!
```

---
layout: post
title: "14th Feb 2020: IAC Team Status Update"
author: criskoduck
categories:
  - team
  - status
tags:
  - module updates
  - tooling
  - personal
  - honeycomb
  - metrics
  - litmus
  - windows
  - powershell
---

The IAC team has been working on a lot of interesting things over the past week. Here's a brief overview of what we've been up to:

## Updates to Puppet Modules

> We've released an update to one module since last week's update: 

Reviewer 2: ```suggestion
We've released an update to two modules since last week: 
```

Author: Good spot

> - [puppetlabs/pwshlib](https://forge.puppet.com/puppetlabs/pwshlib) (v0.4.1)

Reviewer 2: Could probably use a brief overview for this in addition to scheduled task? Maybe just "to fix a bug causing runtime errors on older versions of ruby"?
